### ISSUE TITLE
2.4.2: build fails

### ISSUE THREAD

**Issue by kloczek** [onqtam/doctest#450]
gcc version 11.0.0 20201204.
Sourrce code configure optios:
```
%cmake \
        -D DOCTEST_WITH_MAIN_IN_STATIC_LIB=OFF \
        -D DOCTEST_WITH_TESTS=ON \
```
and build fails:
```
[tkloczko@barrel x86_64-redhat-linux-gnu]$ make
/usr/bin/cmake -S/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2 -B/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu --check-build-system CMakeFiles/Makefile.cmake 0
/usr/bin/cmake -E cmake_progress_start /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu/CMakeFiles /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu//CMakeFiles/progress.marks
make  -f CMakeFiles/Makefile2 all
make[1]: Entering directory '/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu'
make  -f CMakeFiles/assemble_single_header.dir/build.make CMakeFiles/assemble_single_header.dir/depend
make[2]: Entering directory '/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu'
cd /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2 /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2 /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu/CMakeFiles/assemble_single_header.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu'
make  -f CMakeFiles/assemble_single_header.dir/build.make CMakeFiles/assemble_single_header.dir/build
make[2]: Entering directory '/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu'
make[2]: Nothing to be done for 'CMakeFiles/assemble_single_header.dir/build'.
make[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu'
[  2%] Built target assemble_single_header
make  -f examples/all_features/CMakeFiles/disabled.dir/build.make examples/all_features/CMakeFiles/disabled.dir/depend
make[2]: Entering directory '/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu'
cd /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu && /usr/bin/cmake -E cmake_depends "Unix Makefiles" /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2 /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/examples/all_features /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu/examples/all_features /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu/examples/all_features/CMakeFiles/disabled.dir/DependInfo.cmake --color=
make[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu'
make  -f examples/all_features/CMakeFiles/disabled.dir/build.make examples/all_features/CMakeFiles/disabled.dir/build
make[2]: Entering directory '/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu'
[  4%] Linking CXX executable disabled
cd /home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu/examples/all_features && /usr/bin/cmake -E cmake_link_script CMakeFiles/disabled.dir/link.txt --verbose=1
/usr/bin/g++ -O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fstack-protector-strong -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1 -m64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -flto=auto -flto-partition=none -fdata-sections -ffunction-sections -Werror -fstrict-aliasing -pedantic -pedantic-errors -fvisibility=hidden -Wall -Wextra -fdiagnostics-show-option -Wconversion -Wold-style-cast -Wfloat-equal -Wlogical-op -Wundef -Wredundant-decls -Wshadow -Wstrict-overflow=5 -Wwrite-strings -Wpointer-arith -Wcast-qual -Wformat=2 -Wswitch-default -Wmissing-include-dirs -Wcast-align -Wswitch-enum -Wnon-virtual-dtor -Wctor-dtor-privacy -Wsign-conversion -Wdisabled-optimization -Weffc++ -Winvalid-pch -Wmissing-declarations -Woverloaded-virtual -Wunused-but-set-variable -Wunused-result -Wnoexcept -Wdouble-promotion -Wtrampolines -Wzero-as-null-pointer-constant -Wuseless-cast -Wvector-operation-performance -Wshift-overflow=2 -Wnull-dereference -Wduplicated-cond -Walloc-zero -Walloca -Wduplicated-branches -Wcast-align=strict -O2 -g -DNDEBUG -Wl,-z,relro -Wl,--as-needed  -Wl,-z,now -specs=/usr/lib/rpm/redhat/redhat-hardened-ld -flto=auto -flto-partition=none -fuse-linker-plugin -Wl,--gc-sections -rdynamic CMakeFiles/disabled.dir/main.cpp.o CMakeFiles/disabled.dir/alternative_macros.cpp.o CMakeFiles/disabled.dir/assertion_macros.cpp.o CMakeFiles/disabled.dir/stringification.cpp.o CMakeFiles/disabled.dir/reporters_and_listeners.cpp.o CMakeFiles/disabled.dir/subcases.cpp.o CMakeFiles/disabled.dir/logging.cpp.o CMakeFiles/disabled.dir/templated_test_cases.cpp.o CMakeFiles/disabled.dir/test_cases_and_suites.cpp.o CMakeFiles/disabled.dir/asserts_used_outside_of_tests.cpp.o CMakeFiles/disabled.dir/enums.cpp.o CMakeFiles/disabled.dir/concurrency.cpp.o CMakeFiles/disabled.dir/__/__/scripts/coverage_maxout.cpp.o -o disabled  -lpthread
/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/examples/all_features/asserts_used_outside_of_tests.cpp: In function ‘handler’:
/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/doctest/parts/doctest.cpp:669:29: error: null pointer dereference [-Werror=null-dereference]
  669 |     if(getContextOptions()->no_path_in_filenames) {
      |                             ^
lto1: all warnings being treated as errors
lto-wrapper: fatal error: /usr/bin/g++ returned 1 exit status
compilation terminated.
/usr/bin/ld: error: lto-wrapper failed
collect2: error: ld returned 1 exit status
make[2]: *** [examples/all_features/CMakeFiles/disabled.dir/build.make:286: examples/all_features/disabled] Error 1
make[2]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu'
make[1]: *** [CMakeFiles/Makefile2:241: examples/all_features/CMakeFiles/disabled.dir/all] Error 2
make[1]: Leaving directory '/home/tkloczko/rpmbuild/BUILD/doctest-2.4.2/x86_64-redhat-linux-gnu'
make: *** [Makefile:163: all] Error 2
```

**Comment by onqtam** [comment]
This is because of link-time optimizations - a function which indeed does dereference a null pointer gets analyzed in a context outside of doctest.cpp and the silencing of the warning no longer works. However, that null pointer dereference cannot be triggered by a normal code path when doctest is disabled entirely in the project (unless someone starts using things from the `doctest` namespace for things unrelated to unit testing).

I'll fix this tomorrow, thanks for reporting - will think of a better solution. I wonder if the problem would go away if you add `DOCTEST_NOINLINE` to `const ContextOptions* getContextOptions()` in `doctest.cpp`...?

**Comment by dankamongmen** [comment]
```
diff --git doctest/parts/doctest.cpp doctest/parts/doctest.cpp
index 67122e0..c308873 100644
--- doctest/parts/doctest.cpp
+++ doctest/parts/doctest.cpp
@@ -666,6 +666,7 @@ DOCTEST_CLANG_SUPPRESS_WARNING_WITH_PUSH("-Wnull-dereference")
 DOCTEST_GCC_SUPPRESS_WARNING_WITH_PUSH("-Wnull-dereference")
 // depending on the current options this will remove the path of filenames
 const char* skipPathFromFilename(const char* file) {
+#ifndef DOCTEST_CONFIG_DISABLE
     if(getContextOptions()->no_path_in_filenames) {
         auto back    = std::strrchr(file, '\\');
         auto forward = std::strrchr(file, '/');
@@ -675,6 +676,7 @@ const char* skipPathFromFilename(const char* file) {
             return forward + 1;
         }
     }
+#endif
     return file;
 }
 DOCTEST_CLANG_SUPPRESS_WARNING_POP
```

wouldn't this work?

i'm running into this problem attempting to package doctest 2.4.2 for fedora core:

https://kojipkgs.fedoraproject.org//work/tasks/4860/57554860/build.log

**Comment by onqtam** [comment]
@dankamongmen I guess that's a good enough solution - did you try it? Does it work?

**Comment by dankamongmen** [comment]
It solved my issues on Fedora, though on my Debian Unstable machine, I'm seeing:

```
[ 95%] Building CXX object examples/mpi/CMakeFiles/test_mpi.dir/mpi.cpp.o
In file included from /home/dank/src/dankamongmen/doctest/scripts/development_only/doctest/doctest.h:1,
                 from /home/dank/src/dankamongmen/doctest/doctest/extensions/doctest_mpi.h:13,
                 from /home/dank/src/dankamongmen/doctest/examples/mpi/mpi.cpp:1:
/home/dank/src/dankamongmen/doctest/doctest/extensions/doctest_mpi.h: In constructor ‘doctest::mpi_sub_comm::mpi_sub_comm(int)’:
/home/dank/src/dankamongmen/doctest/doctest/extensions/doctest_mpi.h:44:38: error: no match for ‘operator+’ (operand types are ‘doctest::detail::MessageBuilder’ and ‘std::string’ {aka ‘std::__cxx11::basic_string<char>’})
   44 |           "Unable to run test: need "+std::to_string(nb_procs) + " procs"
      |                                      ^~~~~~~~~~~~~~~~~~~~~~~~~
      |                                                     |
      |                                                     std::string {aka std::__cxx11::basic_string<char>}
/home/dank/src/dankamongmen/doctest/doctest/parts/doctest_fwd.h:1996:14: note: in definition of macro ‘DOCTEST_ADD_AT_IMPL’
 1996 |         mb * __VA_ARGS__;                                                                          \
      |              ^~~~~~~~~~~

```

though this seems to be unrelated to my change (i see it on a fresh checkout). haven't dug into this yet.

**Comment by onqtam** [comment]
@dankamongmen can you try with this line: `std::string logfile_name = std::string("doctest_") + std::to_string(rank) + ".log";`?

**Comment by onqtam** [comment]
@dankamongmen can you change this in `doctest_mpi.h`:
```c++
        MESSAGE(
          "Unable to run test: need "+std::to_string(nb_procs) + " procs"
          + " but program launched with only "+std::to_string(doctest::mpi_world_nb_procs()) + "."
        );
```
to this:
```c++
        MESSAGE(
          "Unable to run test: need ", std::to_string(nb_procs), " procs", 
          " but program launched with only ", std::to_string(doctest::mpi_world_nb_procs()), "."
        );
```
and let me know if it fixes the problem?

**Comment by dankamongmen** [comment]
yep, it works. i've got a PR if you want one

**Comment by onqtam** [comment]
@dankamongmen I just pushed the fixes from my local machine & released `2.4.3` - thanks for the help! I guess we should close that PR

**Comment by kloczek** [comment]
Just tested 2.4.3 and it works now :)

Thx
```
+ /usr/bin/make -O -j48 V=1 VERBOSE=1 -C x86_64-redhat-linux-gnu test ARGS=--output-on-failure
make: Entering directory '/home/tkloczko/rpmbuild/BUILD/doctest-2.4.3/x86_64-redhat-linux-gnu'
Running tests...
/usr/bin/ctest --force-new-ctest-process --output-on-failure
Test project /home/tkloczko/rpmbuild/BUILD/doctest-2.4.3/x86_64-redhat-linux-gnu
        Start   1: main.cpp
  1/108 Test   #1: main.cpp ..................................   Passed    0.04 sec
        Start   2: main.cpp_xml
  2/108 Test   #2: main.cpp_xml ..............................   Passed    0.04 sec
        Start   3: main.cpp_junit
  3/108 Test   #3: main.cpp_junit ............................   Passed    0.04 sec
        Start   4: doctest_proxy.h
  4/108 Test   #4: doctest_proxy.h ...........................   Passed    0.04 sec
        Start   5: doctest_proxy.h_xml
  5/108 Test   #5: doctest_proxy.h_xml .......................   Passed    0.04 sec
        Start   6: doctest_proxy.h_junit
  6/108 Test   #6: doctest_proxy.h_junit .....................   Passed    0.04 sec
        Start   7: header.h
  7/108 Test   #7: header.h ..................................   Passed    0.04 sec
        Start   8: header.h_xml
  8/108 Test   #8: header.h_xml ..............................   Passed    0.03 sec
        Start   9: header.h_junit
  9/108 Test   #9: header.h_junit ............................   Passed    0.04 sec
        Start  10: alternative_macros.cpp
 10/108 Test  #10: alternative_macros.cpp ....................   Passed    0.05 sec
        Start  11: alternative_macros.cpp_xml
 11/108 Test  #11: alternative_macros.cpp_xml ................   Passed    0.04 sec
        Start  12: alternative_macros.cpp_junit
 12/108 Test  #12: alternative_macros.cpp_junit ..............   Passed    0.04 sec
        Start  13: assertion_macros.cpp
 13/108 Test  #13: assertion_macros.cpp ......................   Passed    0.04 sec
        Start  14: assertion_macros.cpp_xml
 14/108 Test  #14: assertion_macros.cpp_xml ..................   Passed    0.05 sec
        Start  15: assertion_macros.cpp_junit
 15/108 Test  #15: assertion_macros.cpp_junit ................   Passed    0.04 sec
        Start  16: stringification.cpp
 16/108 Test  #16: stringification.cpp .......................   Passed    0.04 sec
        Start  17: stringification.cpp_xml
 17/108 Test  #17: stringification.cpp_xml ...................   Passed    0.04 sec
        Start  18: stringification.cpp_junit
 18/108 Test  #18: stringification.cpp_junit .................   Passed    0.04 sec
        Start  19: reporters_and_listeners.cpp
 19/108 Test  #19: reporters_and_listeners.cpp ...............   Passed    0.04 sec
        Start  20: reporters_and_listeners.cpp_xml
 20/108 Test  #20: reporters_and_listeners.cpp_xml ...........   Passed    0.04 sec
        Start  21: reporters_and_listeners.cpp_junit
 21/108 Test  #21: reporters_and_listeners.cpp_junit .........   Passed    0.04 sec
        Start  22: subcases.cpp
 22/108 Test  #22: subcases.cpp ..............................   Passed    0.04 sec
        Start  23: subcases.cpp_xml
 23/108 Test  #23: subcases.cpp_xml ..........................   Passed    0.04 sec
        Start  24: subcases.cpp_junit
 24/108 Test  #24: subcases.cpp_junit ........................   Passed    0.04 sec
        Start  25: logging.cpp
 25/108 Test  #25: logging.cpp ...............................   Passed    0.04 sec
        Start  26: logging.cpp_xml
 26/108 Test  #26: logging.cpp_xml ...........................   Passed    0.04 sec
        Start  27: logging.cpp_junit
 27/108 Test  #27: logging.cpp_junit .........................   Passed    0.04 sec
        Start  28: templated_test_cases.cpp
 28/108 Test  #28: templated_test_cases.cpp ..................   Passed    0.04 sec
        Start  29: templated_test_cases.cpp_xml
 29/108 Test  #29: templated_test_cases.cpp_xml ..............   Passed    0.04 sec
        Start  30: templated_test_cases.cpp_junit
 30/108 Test  #30: templated_test_cases.cpp_junit ............   Passed    0.04 sec
        Start  31: test_cases_and_suites.cpp
 31/108 Test  #31: test_cases_and_suites.cpp .................   Passed    0.04 sec
        Start  32: test_cases_and_suites.cpp_xml
 32/108 Test  #32: test_cases_and_suites.cpp_xml .............   Passed    0.04 sec
        Start  33: test_cases_and_suites.cpp_junit
 33/108 Test  #33: test_cases_and_suites.cpp_junit ...........   Passed    0.04 sec
        Start  34: asserts_used_outside_of_tests.cpp
 34/108 Test  #34: asserts_used_outside_of_tests.cpp .........   Passed    0.04 sec
        Start  35: asserts_used_outside_of_tests.cpp_xml
 35/108 Test  #35: asserts_used_outside_of_tests.cpp_xml .....   Passed    0.04 sec
        Start  36: asserts_used_outside_of_tests.cpp_junit
 36/108 Test  #36: asserts_used_outside_of_tests.cpp_junit ...   Passed    0.04 sec
        Start  37: enums.cpp
 37/108 Test  #37: enums.cpp .................................   Passed    0.04 sec
        Start  38: enums.cpp_xml
 38/108 Test  #38: enums.cpp_xml .............................   Passed    0.04 sec
        Start  39: enums.cpp_junit
 39/108 Test  #39: enums.cpp_junit ...........................   Passed    0.03 sec
        Start  40: concurrency.cpp
 40/108 Test  #40: concurrency.cpp ...........................   Passed    0.02 sec
        Start  41: concurrency.cpp_xml
 41/108 Test  #41: concurrency.cpp_xml .......................   Passed    0.02 sec
        Start  42: concurrency.cpp_junit
 42/108 Test  #42: concurrency.cpp_junit .....................   Passed    0.02 sec
        Start  43: coverage_maxout.cpp
 43/108 Test  #43: coverage_maxout.cpp .......................   Passed    0.03 sec
        Start  44: coverage_maxout.cpp_xml
 44/108 Test  #44: coverage_maxout.cpp_xml ...................   Passed    0.04 sec
        Start  45: coverage_maxout.cpp_junit
 45/108 Test  #45: coverage_maxout.cpp_junit .................   Passed    0.04 sec
        Start  46: version
 46/108 Test  #46: version ...................................   Passed    0.04 sec
        Start  47: version_xml
 47/108 Test  #47: version_xml ...............................   Passed    0.04 sec
        Start  48: version_junit
 48/108 Test  #48: version_junit .............................   Passed    0.04 sec
        Start  49: help
 49/108 Test  #49: help ......................................   Passed    0.04 sec
        Start  50: help_xml
 50/108 Test  #50: help_xml ..................................   Passed    0.04 sec
        Start  51: help_junit
 51/108 Test  #51: help_junit ................................   Passed    0.04 sec
        Start  52: outfile
 52/108 Test  #52: outfile ...................................   Passed    0.02 sec
        Start  53: outfile_xml
 53/108 Test  #53: outfile_xml ...............................   Passed    0.02 sec
        Start  54: outfile_junit
 54/108 Test  #54: outfile_junit .............................   Passed    0.02 sec
        Start  55: count
 55/108 Test  #55: count .....................................   Passed    0.03 sec
        Start  56: count_xml
 56/108 Test  #56: count_xml .................................   Passed    0.04 sec
        Start  57: count_junit
 57/108 Test  #57: count_junit ...............................   Passed    0.04 sec
        Start  58: list_test_cases
 58/108 Test  #58: list_test_cases ...........................   Passed    0.04 sec
        Start  59: list_test_cases_xml
 59/108 Test  #59: list_test_cases_xml .......................   Passed    0.04 sec
        Start  60: list_test_cases_junit
 60/108 Test  #60: list_test_cases_junit .....................   Passed    0.04 sec
        Start  61: list_test_suites
 61/108 Test  #61: list_test_suites ..........................   Passed    0.04 sec
        Start  62: list_test_suites_xml
 62/108 Test  #62: list_test_suites_xml ......................   Passed    0.04 sec
        Start  63: list_test_suites_junit
 63/108 Test  #63: list_test_suites_junit ....................   Passed    0.04 sec
        Start  64: list_reporters
 64/108 Test  #64: list_reporters ............................   Passed    0.04 sec
        Start  65: list_reporters_xml
 65/108 Test  #65: list_reporters_xml ........................   Passed    0.04 sec
        Start  66: list_reporters_junit
 66/108 Test  #66: list_reporters_junit ......................   Passed    0.04 sec
        Start  67: all_binary
 67/108 Test  #67: all_binary ................................   Passed    0.04 sec
        Start  68: all_binary_xml
 68/108 Test  #68: all_binary_xml ............................   Passed    0.04 sec
        Start  69: all_binary_junit
 69/108 Test  #69: all_binary_junit ..........................   Passed    0.04 sec
        Start  70: abort_after
 70/108 Test  #70: abort_after ...............................   Passed    0.04 sec
        Start  71: abort_after_xml
 71/108 Test  #71: abort_after_xml ...........................   Passed    0.04 sec
        Start  72: abort_after_junit
 72/108 Test  #72: abort_after_junit .........................   Passed    0.04 sec
        Start  73: first_last
 73/108 Test  #73: first_last ................................   Passed    0.04 sec
        Start  74: first_last_xml
 74/108 Test  #74: first_last_xml ............................   Passed    0.04 sec
        Start  75: first_last_junit
 75/108 Test  #75: first_last_junit ..........................   Passed    0.04 sec
        Start  76: filter_1
 76/108 Test  #76: filter_1 ..................................   Passed    0.04 sec
        Start  77: filter_1_xml
 77/108 Test  #77: filter_1_xml ..............................   Passed    0.04 sec
        Start  78: filter_1_junit
 78/108 Test  #78: filter_1_junit ............................   Passed    0.04 sec
        Start  79: filter_2
 79/108 Test  #79: filter_2 ..................................   Passed    0.04 sec
        Start  80: filter_2_xml
 80/108 Test  #80: filter_2_xml ..............................   Passed    0.04 sec
        Start  81: filter_2_junit
 81/108 Test  #81: filter_2_junit ............................   Passed    0.04 sec
        Start  82: filter_3
 82/108 Test  #82: filter_3 ..................................   Passed    0.04 sec
        Start  83: filter_3_xml
 83/108 Test  #83: filter_3_xml ..............................   Passed    0.04 sec
        Start  84: filter_3_junit
 84/108 Test  #84: filter_3_junit ............................   Passed    0.04 sec
        Start  85: order_1
 85/108 Test  #85: order_1 ...................................   Passed    0.04 sec
        Start  86: order_1_xml
 86/108 Test  #86: order_1_xml ...............................   Passed    0.04 sec
        Start  87: order_1_junit
 87/108 Test  #87: order_1_junit .............................   Passed    0.04 sec
        Start  88: order_2
 88/108 Test  #88: order_2 ...................................   Passed    0.04 sec
        Start  89: order_2_xml
 89/108 Test  #89: order_2_xml ...............................   Passed    0.04 sec
        Start  90: order_2_junit
 90/108 Test  #90: order_2_junit .............................   Passed    0.05 sec
        Start  91: order_3
 91/108 Test  #91: order_3 ...................................   Passed    0.03 sec
        Start  92: order_3_xml
 92/108 Test  #92: order_3_xml ...............................   Passed    0.04 sec
        Start  93: order_3_junit
 93/108 Test  #93: order_3_junit .............................   Passed    0.04 sec
        Start  94: disabled
 94/108 Test  #94: disabled ..................................   Passed    0.04 sec
        Start  95: disabled_xml
 95/108 Test  #95: disabled_xml ..............................   Passed    0.04 sec
        Start  96: disabled_junit
 96/108 Test  #96: disabled_junit ............................   Passed    0.04 sec
        Start  97: exe_with_static_libs
 97/108 Test  #97: exe_with_static_libs ......................   Passed    0.04 sec
        Start  98: exe_with_static_libs_xml
 98/108 Test  #98: exe_with_static_libs_xml ..................   Passed    0.04 sec
        Start  99: exe_with_static_libs_junit
 99/108 Test  #99: exe_with_static_libs_junit ................   Passed    0.04 sec
        Start 100: executable_dll_and_plugin
100/108 Test #100: executable_dll_and_plugin .................   Passed    0.04 sec
        Start 101: executable_dll_and_plugin_xml
101/108 Test #101: executable_dll_and_plugin_xml .............   Passed    0.05 sec
        Start 102: executable_dll_and_plugin_junit
102/108 Test #102: executable_dll_and_plugin_junit ...........   Passed    0.04 sec
        Start 103: playground
103/108 Test #103: playground ................................   Passed    0.02 sec
        Start 104: playground_xml
104/108 Test #104: playground_xml ............................   Passed    0.02 sec
        Start 105: playground_junit
105/108 Test #105: playground_junit ..........................   Passed    0.02 sec
        Start 106: test_mpi
106/108 Test #106: test_mpi ..................................   Passed    0.85 sec
        Start 107: test_mpi_xml
107/108 Test #107: test_mpi_xml ..............................   Passed    0.75 sec
        Start 108: test_mpi_junit
108/108 Test #108: test_mpi_junit ............................   Passed    0.77 sec

100% tests passed, 0 tests failed out of 108

Total Test time (real) =   6.57 sec
```
